src/main/resources/templates/scala-lang/scalaClassGraphqlResponseProjection.ftl

```diff
@@ -26,10 +29,12 @@ import java.util.Objects
 </#list>
 class ${className} extends GraphQLResponseProjection {
 
-<#if fields?has_content>
-    override def all$(): ${className} = all$(${responseProjectionMaxDepth})
+<#if fields?has_content && generateAllMethodInProjection>
+    var projectionDepthOnFields = new HashMap[String, Int]
```

Comment:
please use `val`, because javamap is mutable struction.

Reply:
I changed it to val en the scala native hashmap with getOrElse